src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -153,10 +153,10 @@ private void consume() throws Exception {
         _sensors._consumeError.record();
         continue;
       }
-      int partition = record.partition();
-      long index = (Long) avroRecord.get(DefaultTopicSchema.INDEX_FIELD.name());
+      int partition = (int) avroRecord.get(DefaultTopicSchema.PARTITION_FIELD.name());
```

Comment:
@lindong28 @smccauliff We need to key the messages regardless to overcome the lack of partition identity. Rather than change the DefaultTopicSchema, could we simply store the expected index by message key? record.key()